drivers/sensor/lis3mdl/lis3mdl_trigger.c

```diff
@@ -58,12 +55,11 @@ static void lis3mdl_gpio_callback(const struct device *dev,
 {
 	struct lis3mdl_data *drv_data =
 		CONTAINER_OF(cb, struct lis3mdl_data, gpio_cb);
+	const struct lis3mdl_config *config = dev->config;
```

Comment:
`dev` is a GPIO device here, not a sensor device

Reply:
Done!